### PR TITLE
⭐️ SBOM connection

### PR DESCRIPTION
**Pull request by chris-rock**
This PR adds new support to interact with sbom files:

```
cnquery sbom docker alpine:3.19 --output cyclonedx-json
cnquery shell sbom cyclonedx-json
```

Or you can just pipe content:

```
cnquery sbom docker alpine:3.19 --output cyclonedx-json |  cnquery shell sbom - -c "packages"  
```

To inspect sbom from docker hub:

```
docker buildx imagetools inspect mondoo/client --format "{{json .SBOM }}"  |  jq '."linux/amd64"."SPDX"' |  cnquery shell sbom -
````
> Note that docker stores multiple sboms in the manifest, eg. for each architecture and potentially in different formats. Therefore we need to extract the individual sbom first.

TODO:

- [x] update asset resource
- [ ] container / device id (plan to do updates in follow-up PRs)

### CODE REVIEW

**Comment by tas50** [comment]
@chris-rock Here's an example docker command to fetch a SBOM from the container registry image manifest:

```
docker buildx imagetools inspect mondoo/client --format "{{json .SBOM }}"
```

File: sbom/cyclonedx.go [hunk 1]
```diff
@@ -105,12 +120,118 @@ func (ccx *CycloneDX) convert(bom *Sbom) (*cyclonedx.BOM, error) {
...
+
+var familyMap = map[string][]string{
+	"windows": []string{"windows", "os"},
+	"debian":  []string{"linux", "unix", "os"},
+	"ubuntu":  []string{"linux", "unix", "os"},
+	"centos":  []string{"linux", "unix", "os"},
+	"alpine":  []string{"linux", "unix", "os"},
+	"fedora":  []string{"linux", "unix", "os"},
+	"rhel":    []string{"linux", "unix", "os"},
+}
```
Comment (scottford-io) [inline sbom/cyclonedx.go]:
```suggestion
var familyMap = map[string][]string{
	"windows": []string{"windows", "os"},
	"macos":   []string{"darwin", "bsd", "unix", "os"},
	"debian":  []string{"linux", "unix", "os"},
	"ubuntu":  []string{"linux", "unix", "os"},
	"centos":  []string{"linux", "unix", "os"},
	"alpine":  []string{"linux", "unix", "os"},
	"fedora":  []string{"linux", "unix", "os"},
	"rhel":    []string{"linux", "unix", "os"},
}
```

Comment (scottford-io) [reply sbom/cyclonedx.go]: @chris-rock I think this is right... 🤔

**Comment by scottford-io** [comment]
@chris-rock I tested inspecting a remote SBOM on Docker Hub. When I enter the `cnquery shell` and run `packages`, I get all sorts of `rpc` errors:

```
packages {*}
21 errors occurred:
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/musl@1.2.5-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/musl-utils@1.2.5-r0?arch=x86_64&upstream=musl&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/alpine-baselayout@3.6.5-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/alpine-keys@2.4-r1?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/ca-certificates@20240226-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libidn2@2.3.7-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libssl3@3.3.0-r2?arch=x86_64&upstream=openssl&distro=alpine-3.20.0) doesn't exist
	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libunistring@1.2-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
	```

**Comment by chris-rock** [comment]
> @chris-rock I tested inspecting a remote SBOM on Docker Hub. When I enter the `cnquery shell` and run `packages`, I get all sorts of `rpc` errors:
> 
> ```
> packages {*}
> 21 errors occurred:
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/musl@1.2.5-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/musl-utils@1.2.5-r0?arch=x86_64&upstream=musl&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/alpine-baselayout@3.6.5-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/alpine-keys@2.4-r1?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/ca-certificates@20240226-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libidn2@2.3.7-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libssl3@3.3.0-r2?arch=x86_64&upstream=openssl&distro=alpine-3.20.0) doesn't exist
> 	* rpc error: code = Unknown desc = resource 'package' (id: pkg:apk/alpine/libunistring@1.2-r0?arch=x86_64&distro=alpine-3.20.0) doesn't exist
> 	```
> ```

Can you share the images that you tested?

**Comment by scottford-io** [comment]
@chris-rock I ran the example in your PR above ☝️ 

```
docker buildx imagetools inspect mondoo/client --format "{{json .SBOM }}"  |  jq '."linux/amd64"."SPDX"' |  cnquery shell sbom -
```

**Pull request merged**